tensor2tensor/data_generators/problem.py

```diff
@@ -842,6 +842,54 @@ def _dataset_partition(self, mode, config, hvd=None):
     assert partition_id < num_partitions
     return partition_id, num_partitions
 
+  def apply_batch_settings(self, dataset, hparams, num_shards, num_threads,
+                           is_training, **kwargs):
+    """Applies padding and various other setup tasks to dataset for batching."""
+
+    max_length = self.max_length(hparams)
+
+    def gpu_valid_size(example):
+        drop_long_sequences = is_training or hparams.eval_drop_long_sequences
+        return data_reader.example_valid_size(example, hparams.min_length,
+                                              max_length
+                                              if drop_long_sequences else 10 ** 9)
+    # for unpacked datasets, bucket by length
+    dataset = dataset.filter(gpu_valid_size)
+    batching_scheme = self._get_batching_scheme(hparams, num_shards)
+
+    # if unpacked (variable length sequences) and we are chunking
+    # input features, we need to pad the features that we
+    # chunk to the next neares multiple of the chunk length,
+    if hasattr(hparams, 'chunk_length'):
```

Comment:
not your fault, but while we are here, this code block can be moved out:

```suggestion
    # Fathom
    # chunking support
    # comments etc.
    dataset = maybe_prepare_for_chunking(hparams, dataset)
```

and `maybe_prepare_for_chunking` live alongside https://github.com/medicode/diseaseTools/blob/c470e6e8a3f8e25328ff8a76a694b7a083ab6846/fathomt2t_dependencies/common_t2t_utils.py#L108

Reply:
Fixed